shared/formatS3key.test.ts

```diff
@@ -0,0 +1,19 @@
+import { formatS3Key } from "./formatS3key";
+
+describe('formatS3Key', () => {
+    it('should format the S3 key with the external ID and .json extension', () => {
+        const externalId = 'test-external-id';
+        const expectedKey = 'test-external-id.json';
+        expect(formatS3Key(externalId)).toBe(expectedKey);
+    });
+    it('should replace spaces with a hyphen', () => {
+        const externalId = 'test external id';
+        const expectedKey = 'test-external-id.json';
+        expect(formatS3Key(externalId)).toBe(expectedKey);
+    });
+    it('should replace forward slashes with a hyphen', () => {
+        const externalId = 'test/external/id';
+        const expectedKey = 'test-external-id.json';
+        expect(formatS3Key(externalId)).toBe(expectedKey);
```

Comment:
Does this run the risk that some stories might overwrite each other in S3? if we had 3 files arrive with each of these external ids, then we'd be left with only the most recent, even though they've each got different ids. (Hopefully actually we still have all 3, but only as versions of object with the same key). 
Do we have a concrete problem with the spaced or slashed keys? Maybe we run the id through a url encoder instead?